src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/VariantsSparkSink.java

```diff
@@ -0,0 +1,150 @@
+            final VCFHeader header) throws IOException {
+
+        final Configuration conf = ctx.hadoopConfiguration();
+        if (outputFile.endsWith(BGZFCodec.DEFAULT_EXTENSION)) {
+            conf.setBoolean(FileOutputFormat.COMPRESS, true);
+            conf.setClass(FileOutputFormat.COMPRESS_CODEC, BGZFCodec.class, CompressionCodec.class);
+        } else {
+            conf.setBoolean(FileOutputFormat.COMPRESS, false);
+        }
+
+        final String outputPartsDirectory = outputFile + ".parts";
+        saveAsShardedHadoopFiles(ctx, conf, outputPartsDirectory, variants,  header, false);
+        VCFFileMerger.mergeParts(outputPartsDirectory, outputFile, header);
+    }
+
+    private static void saveAsShardedHadoopFiles(
+            final JavaSparkContext ctx, final Configuration conf, final String outputFile, JavaRDD<VariantContext> variants,
+            final VCFHeader header, final boolean writeHeader) throws IOException {
+        // Set the static header on the driver thread.
+        SparkVCFOutputFormat.setVCFHeader(header);
+
+        final Broadcast<VCFHeader> headerBroadcast = ctx.broadcast(header);
+
+        // SparkVCFOutputFormat is a static class, so we need to copy the header to each worker then call
+        final JavaRDD<VariantContext> variantsRDD = setHeaderForEachPartition(variants, headerBroadcast);
+
+        // The expected format for writing is JavaPairRDD where the key is ignored and the value is VariantContextWritable.
+        final JavaPairRDD<VariantContext, VariantContextWritable> rddVariantContextWriteable = pairVariantsWithVariantContextWritables(variantsRDD);
+
+        rddVariantContextWriteable.saveAsNewAPIHadoopFile(outputFile, VariantContext.class, VariantContextWritable.class, getOutputFormat(writeHeader), conf);
+    }
+
+    private static JavaRDD<VariantContext> setHeaderForEachPartition(final JavaRDD<VariantContext> variants, final Broadcast<VCFHeader> headerBroadcast) {
+        return variants.mapPartitions(readIterator -> {
+            SparkVCFOutputFormat.setVCFHeader(headerBroadcast.getValue());
+            return new IteratorIterable<>(readIterator);
+        });
+    }
+
+    private static JavaPairRDD<VariantContext, VariantContextWritable> pairVariantsWithVariantContextWritables(JavaRDD<VariantContext> records) {
+        return records.mapToPair(variantContext -> {
+            final VariantContextWritable variantContextWritable = new VariantContextWritable();
+            variantContextWritable.set(variantContext);
+            return new Tuple2<>(variantContext, variantContextWritable);
+        });
+    }
+
+    private static Class<? extends OutputFormat<NullWritable, VariantContextWritable>> getOutputFormat(final boolean writeHeader) {
+        return writeHeader ? SparkVCFOutputFormat.class : SparkHeaderlessVCFOutputFormat.class;
+    }
+
+    private static String makeFilePathAbsolute(String path){
```

Comment:
These are proliferating.  Can we put this in BucketUtils or IOUtils?  There's something similar in `FeatureInput.makeIntoAbsolutePath` that does something special for TILEDB uris.